azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -776,6 +802,26 @@ public void deleteExecutionDirectory() {
     }
   }
 
+  private Set<Pair<Integer, Integer>> getActiveProjectVersions() {
+    final Set<Pair<Integer, Integer>> activeProjectVersions = new HashSet<>();
+    for (final FlowRunner runner : FlowRunnerManager.this.runningFlows.values()) {
+      final ExecutableFlow flow = runner.getExecutableFlow();
+      activeProjectVersions.add(new Pair<>(flow
+          .getProjectId(), flow.getVersion()));
+    }
+    return activeProjectVersions;
+  }
+
+  /**
+   * Checks if the project version contains any running flow
+   */
+  private boolean isActiveProject(final ProjectVersion version) {
+    final Pair<Integer, Integer> versionKey = new Pair<>(version.getProjectId(),
+        version.getVersion());
+    return getActiveProjectVersions().contains(versionKey);
+  }
+
+
   private class CleanerThread extends Thread {
```

Comment:
Today LRU cache cannot work immediately on delete, because there's always a hardlink from execution to project directory.

Please add a todo comment to describe the current the situation.

Reply:
thanks, mentioned it in PR.